third_party/kmod/CMakeLists.txt

```diff
@@ -81,3 +80,66 @@ add_custom_command(
         ""
     VERBATIM
 )
+else()
+message(STATUS "Skipping Meltdown kernel module on non-x86/64/ppc CPUs")
+endif()
+
+if (${CMAKE_SYSTEM_PROCESSOR} MATCHES "^aarch64$")
```

Comment:
Copy-pasting 40 lines of build description doesn't seem like the right call here. What other options does CMake give us?

Maybe this is a job for [`function`](https://cmake.org/cmake/help/v3.16/command/function.html)?

Reply:
Do you want me to use the function even across different directories? Should the function be defined in the root CMakeLists.txt file?

Reply:
I admit I don't have a ready sense for the tradeoffs involved here. Do you have a preference?

Reply:
Added a midlevel CMakeLists.txt file with a function for building kernel modules.